apps/main/src/pages/_app.tsx

```diff
@@ -90,7 +90,10 @@ function CurveApp({ Component }: AppProps) {
     const { rLocale } = getLocaleFromUrl()
     const parsedLocale = rLocale?.value ?? 'en'
     initTranslation(i18n, parsedLocale)
-    dynamicActivate(parsedLocale)
+    ;(async () => {
```

Comment:
Is it possible to remove repetitions of this? Just one per app should be enough. Your explanation in the PR description might come in handy, too

Reply:
There is only one invocation per app on load, but sadly we have 4 apps so there's some repetition there.

Then each app also has its own `usePageOnMount` where there's a watch / useEffect to see if the locale has changed. You can argue that should be enough and the call in `_app.tsx` can be removed, but I'm afraid somebody somewhere doesn't use that hook and that specific page would fail to load the locale.

However, I've added documentation explaining the situation to the `dynamicActivate` function.

Reply:
There're multiple invovations per app if I'm not mistaken?

Reply:
Per app there's an invocation in _app.tsx inside a useEffect with [] as it's dependency, so I'm assuming that happens once on initial mount, unless I'm missing something.

Reply:
For example 
https://github.com/curvefi/curve-frontend/blob/b067240955dc2ae2134848760d80150e02a67de8/apps/loan/src/hooks/usePageOnMount.ts#L259

Reply:
That's for locale changes and I'm not sure if every page even uses that hook. If they don't and you'd remove the invocation in _app.tsx you'd have no locale initialization at all. And since we have 4 diff apps that are slightly different I don't want to move that code from the hook to the four _app.tsx's. This needs a refactor but imo is outside the scope of the PR.